matrix/basic_matrix.py

```diff
@@ -189,32 +292,46 @@ def is_hankel(self):
 
         return True
 
+    def is_symplectic(self):
+        """ Checks whether matrix is symplectic.  Returns True if symplectic matrix, False otherwise """
+
+        if not self.is_square():
+            return False
+
+        n = self.len_col // 2
+
+        ls_identity = self.identity_ls_entries(n)
+        ls_identity_neg = [[-1 * i for i in inner] for inner in ls_identity]
+        ls_zeros = self.zero_ls_entries(n, n)
+
+        # block matrix must be non-singular, skew-symmetric
+        # Use the typical choice:
+        # [[0  I_n],
+        # [-I_n 0]]
+        block_matrix = Matrix(self.create_block_ls_entries(top_left=Matrix(ls_zeros), top_right=Matrix(ls_identity),
+                                                           bottom_left=Matrix(ls_identity_neg),
+                                                           bottom_right=Matrix(ls_zeros)))
+
+        transpose = Matrix(ls_entries=[[self[j][i] for j in range(self.len_row)] for i in range(self.len_col)])
```

Comment:
Would it make more sense here to call some sort of matrix transpose method or `self.transpose`?

Reply:
Yeah it would, was running into recursion issues when using the regular self.transpose() method.   So will address this the same way as the _mul private method (using the decorators to control whether to use the default Matrix class for the returned object). _mul and _transpose should only be used in this basic_matrix.py file for the is_something_matrix() checks.  Will make PR for this.